### ISSUE TITLE
CPU-local freelists with restartable sequences

### ISSUE THREAD

**Issue by kentonv** [gperftools/gperftools#1042]
Now that restartable sequences have finally landed in Linux, are there any plans to use them in the public version of tcmalloc?

Is there anything that I or Cloudflare can do to help here? I guess the code has to be ported out of Google's internal branch?

**Comment by alk** [comment]
Yes, I am working on this. Cannot say about timing because I am currently busy with other stuff. And internal version being quite different rcu-wise and everything-wise makes it less trivial work.

I am hoping something might be available for testing within about 4-6 weeks from now.

**Comment by alk** [comment]
While it won't help specific per-cpu caches effort directly, if you can, please consider contributing malloc traces from some of your apps (I plan to share google-internal traces this year too). Instructions here: https://github.com/alk/malloc-trace-replay

**Comment by toddlipcon** [comment]
BTW I came across this article with a hacky way of implementing rseq without the syscall (for compatibility on older kernels): https://www.pvk.ca/Blog/2018/08/25/restartable-sequences-with-the-polysemic-null-segment-selector/
